include/uapi/sound/sof/abi.h

```diff
@@ -59,4 +61,34 @@
 /* SOF ABI magic number "SOF\0". */
 #define SOF_ABI_MAGIC		0x00464F53
 
+#define SOF_MANIFEST_DATA_TYPE_NHLT 1
+
+/**
+ * struct sof_manifest_tlv - SOF manifest TLV data
+ * @type: type of data
+ * @size: data size (not including the size of this struct)
+ * @data: payload data
+ */
+struct sof_manifest_tlv {
+	__u32 type;
+	__u32 size;
```

Comment:
is there also a "super" struct holding these ipc4_tuples? I mean I can use ipc4_tuple but I still need the master struct describing the amount of items etc. or this is just about the location/file of the structs?

Reply:
@ujfalusi so the tlv is in abi.h as the original sof_manifest was in abi.h, I just "augmented it". You're right that the tlv struct is similar/same than ipc4_tuple, but it would just look strange if I would use it as generic manifest data tlv. If there's somewhere else in alsa a generic similar struct to use, that would perhaps make more sense. People would just start to wonder what this got to do with ipc4...

Reply:
@juimonen, there is `struct snd_ctl_tlv` in include/uapi/sound/asound.h, but it is not **packed**, which is needed for us when communicating with the firmware or reading file. It is using `unsigned int`, which is u32 mostly, but for inter-processor communication we should be strict on type and size.
But the concept is not that innovative, I'm sure it is used in lot of places (does IPC3 uses similar construct?). Anyways, if you have a good generic name for it, we can switch to use it in all places in SOF.

Reply:
I think we should reserve this and unify all the TLV structs in SOF as a follow up patch @ujfalusi. FWIW, when we move to only tuples for IPC data, it will be needed anyway

Reply:
Can these be added to the header.h header file where we have `struct sof_abi_hdr` ?